roles/dd-nodepool/files/nodepool_server.py

```diff
@@ -20,24 +20,39 @@
 class NodepoolCheck(AgentCheck):
 
     SERVICE_CHECK_NAME = 'nodepool_server.got_status'
+    IMAGES_CHECK_NAME = 'nodepool_server.got_images'
 
     def check(self, instance):
-        if 'status_url' not in instance:
-            self.log.info("Skipping instance, no url found.")
-            return
-
         status_url = instance['status_url']
+        images_url = instance['images_url']
```

Comment:
you removed the if status_url in instance check, so these will KeyError if the value isn't present in config. Just move it down and use instance['status_url'] instead of status_url in the if block.